src/lrtest.jl

```diff
@@ -106,42 +107,53 @@ function lrtest(mods::StatisticalModel...; atol::Real=0.0)
     end
 
     dev = deviance.(mods)
-    Δdev = _diff(dev)
 
-    Δdf = _diff(df)
+    Δdf = (NaN, _diff(df)...)
     dfr = Int.(dof_residual.(mods))
 
-    if (forward && any(x -> x > 0, Δdev)) || (!forward && any(x -> x < 0, Δdev))
-        throw(ArgumentError("Residual deviance must be strictly lower " *
-                            "in models with more degrees of freedom"))
+    ll = loglikelihood.(mods)
+    chisq = (NaN, 2 .* abs.(_diff(ll))...)
+
+    for i in 2:length(ll)
+        if ((forward && ll[i-1] > ll[i]) ||
+            (!forward && ll[i-1] < ll[i])) &&
+            ll[i-1] ≉ ll[i]
```

Comment:
This was actually part of my earlier comment but got lost because the remainder of the comment was beefy. Should we allow the user to provide a tolerance for this check? We already have an `atol` argument used for checking whether the models are nested, perhaps it would make sense to reuse that here?

Reply:
Yeah why not. I've pushed a commit to do that. BTW I discovered that the condition was backwards!